### DISCUSSION TITLE
自定义组件x-decorator: FormItem; 设置required 为true不生效

### DISCUSSION THREAD

**Discussion by marspal** [alibaba/formily#2681] [Q&A 2.x] [answered]
1.  使用formily numberPicker组件, required是正常的

![image](https://user-images.githubusercontent.com/41223123/147409502-851e05df-6b6d-4548-a540-2b70ad50088f.png)

![image](https://user-images.githubusercontent.com/41223123/147409490-757a32d8-d285-4214-9599-e162c4b455be.png)

2. 使用numberPicker 简单封装，required失效

![image](https://user-images.githubusercontent.com/41223123/147409544-558fc772-39b4-48f3-8e77-4bddac70a1ef.png)

SalaryField 封装
![image](https://user-images.githubusercontent.com/41223123/147409580-54f95965-fb17-4ae4-a120-1c31e7dcf23c.png)

required失效
![image](https://user-images.githubusercontent.com/41223123/147409592-6606638c-5fdb-49db-af57-c519b52a2974.png)

通过formily-devtool查看; required还是false， 请问一下 是对自定义组件有什么过滤嘛？
![image](https://user-images.githubusercontent.com/41223123/147409627-8780d7d9-3494-418b-934a-9254a22f915a.png)

**Comment by marspal** [comment]
这是我的临时 解决方案
![image](https://user-images.githubusercontent.com/41223123/147425895-60665855-aa9c-48dc-acbf-1a05e1744d36.png)

**Comment by janryWang** [comment, marked as answer]
没过滤，FormItem请用formily提供的，不要用组件库的

**Reply by marspal** [reply]
FormItem用的是formily提供的, 文档上有说的; numberpicker用的是antd的，在SalaryField组件里useSchema 能获取到正确的schema文件; 
![image](https://user-images.githubusercontent.com/41223123/147431665-bedcf24b-b3f6-476a-9b3a-49c4b759ea8a.png)
![image](https://user-images.githubusercontent.com/41223123/147431683-3aca65cb-9d36-4971-8e80-7d8c82a4dfa0.png)

**Comment by janryWang** [comment]
给个codesandbox看看吧
